src/goto-programs/vcd_goto_trace.cpp

```diff
@@ -55,7 +55,7 @@ std::string as_vcd_binary(
   }
   else if(expr.id()==ID_union)
   {
-    assert(expr.operands().size()==1);
+    INVARIANT(expr.operands().size() == 1, "union only takes 1 operand");
```

Comment:
I'd suggest to use `union_exprt`/`to_union_expr`. Let's not duplicate invariant checks.